scikeras/wrappers.py

```diff
@@ -998,10 +1060,22 @@ def r_squared(y_true, y_pred):
         Since ScikitLearn's `score` uses R^2 by default, it is
         advisable to use the same loss/metric when optimizing the model.
         """
-        ss_res = k_backend.sum(k_backend.square(y_true - y_pred), axis=0)
-        ss_tot = k_backend.sum(
-            k_backend.square(y_true - k_backend.mean(y_true, axis=0)), axis=0
+        # Ensure input dytpes match
+        dtype_y_true = np.dtype(y_true.dtype.as_numpy_dtype())
+        dtype_y_pred = np.dtype(y_pred.dtype.as_numpy_dtype())
+        dest_dtype = np.promote_types(dtype_y_pred, dtype_y_true)
+        y_true = tf.cast(y_true, dtype=dest_dtype)
+        y_pred = tf.cast(y_pred, dtype=dest_dtype)
```

Comment:
Why is this batch of type-casting necessary? Does the backend calculation of R^2 require it?

Reply:
It requires `y_true` and `y_pred` to match dtypes. We could just cast to `float32` (since I think `y_pred` will always be `float32`) instead of trying to figure out the destination dtype using `np.promote_types`.